spring-ai-core/src/main/java/org/springframework/ai/chat/client/observation/DefaultChatClientObservationConvention.java

```diff
@@ -0,0 +1,144 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.springframework.ai.chat.client.observation;
+
+import java.util.stream.Collectors;
+
+import org.springframework.ai.chat.client.observation.ChatClientObservationDocumentation.LowCardinalityKeyNames;
+import org.springframework.ai.chat.observation.ChatModelObservationDocumentation;
+import org.springframework.lang.Nullable;
+import org.springframework.util.CollectionUtils;
+
+import io.micrometer.common.KeyValue;
+import io.micrometer.common.KeyValues;
+
+/**
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+
+public class DefaultChatClientObservationConvention implements ChatClientObservationConvention {
+
+	public static final String DEFAULT_NAME = "gen_ai.chat.client.operation";
+
+	private static final KeyValue CHAT_CLIENT_TOOL_FUNCTION_CALLBACKS_NONE = KeyValue.of(
+			ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_TOOL_FUNCTION_CALLBACKS,
+			KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_TOOL_FUNCTION_NAMES_NONE = KeyValue.of(
+			ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_TOOL_FUNCTION_NAMES,
+			KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_ADVISOR_NONE = KeyValue
+		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_ADVISORS, KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_ADVISOR_PARAM_NONE = KeyValue
+		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_ADVISOR_PARAMS, KeyValue.NONE_VALUE);
+
+	private final String name;
+
+	public DefaultChatClientObservationConvention() {
+		this(DEFAULT_NAME);
+	}
+
+	public DefaultChatClientObservationConvention(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public String getName() {
+		return this.name;
+	}
+
+	@Override
+	@Nullable
+	public String getContextualName(ChatClientObservationContext context) {
+		return "%s %s".formatted(context.getOperationMetadata().operationType(),
+				context.getOperationMetadata().provider());
+	}
+
+	@Override
+	public KeyValues getLowCardinalityKeyValues(ChatClientObservationContext context) {
+		return KeyValues.of(springAiKind(context), aiOperationType(context), aiProvider(context), stream(context));
+	}
+
+	@Override
+	public KeyValues getHighCardinalityKeyValues(ChatClientObservationContext context) {
+		return KeyValues.of(toolFunctionNames(context), toolFunctionCallbacks(context), chatClientAvisor(context),
+				chatClientAvisorParam(context));
+	}
+
+	protected KeyValue springAiKind(ChatClientObservationContext context) {
+		return KeyValue.of(ChatClientObservationDocumentation.LowCardinalityKeyNames.SPRING_AI_KIND, "chat.client");
```

Comment:
I would rename `chat.client` to `chat_client` because that's the typical naming convention for attribute values of this type (whereas the key itself uses the dots convention)